apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore.IntegrationTests/ErrorReporting/ErrorReportingTest.cs

```diff
@@ -34,12 +34,40 @@ public class ErrorReportingTest
         private readonly ErrorEventEntryPolling _polling = new ErrorEventEntryPolling();
 
         [Fact]
-        public async Task NoExceptions()
+        public async Task NoExceptions_ErrorReporting()
+            => await NoExceptions(
+                new WebHostBuilder().UseStartup<ReportToErrorReportingTestApplication>());
+
+        [Fact]
+        public async Task LogsException_ErrorReporting()
+            => await LogsException(
+                new WebHostBuilder().UseStartup<ReportToErrorReportingTestApplication>());
+
+        [Fact]
+        public async Task LogsMultipleExceptions_ErrorReporting()
+            => await LogsMultipleExceptions(
+                new WebHostBuilder().UseStartup<ReportToErrorReportingTestApplication>());
+
+        [Fact]
+        public async Task NoExceptions_Logging()
+            => await NoExceptions(
+                new WebHostBuilder().UseStartup<ReportToLoggingTestApplication>());
+
+        [Fact]
+        public async Task LogsException_Logging()
+             => await LogsException(
+                new WebHostBuilder().UseStartup<ReportToLoggingTestApplication>());
+
+        [Fact]
+        public async Task LogsMultipleExceptions_Logging()
+            => await LogsMultipleExceptions(
+                new WebHostBuilder().UseStartup<ReportToLoggingTestApplication>());
+
+        public async Task NoExceptions(IWebHostBuilder builder)
         {
             string testId = Utils.GetTestId();
             DateTime startTime = DateTime.UtcNow;
 
-            var builder = new WebHostBuilder().UseStartup<ErrorReportingTestApplication>();
```

Comment:
Again, this can probably be a generic method that uses `T` here (same comment below).